src/Impostor.Server/Http/GamesController.cs

```diff
@@ -0,0 +1,154 @@
+using Impostor.Api.Innersloth;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Options;
+
+namespace Impostor.Server.Http;
+
+/// <summary>
+/// This controller has method to get a list of public games, join by game and create new games.
+/// </summary>
+[Route("/api/games")]
+[ApiController]
+public sealed class GamesController : ControllerBase
+{
+    private readonly IGameManager _gameManager;
+    private readonly ListingManager _listingManager;
+    private readonly HostServer _hostServer;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="GamesController"/> class.
+    /// </summary>
+    /// <param name="gameManager">GameManager containing a list of games.</param>
+    /// <param name="listingManager">ListingManager responsible for filtering.</param>
+    /// <param name="serverConfig">Impostor configuration section containing the public ip address of this server.</param>
+    public GamesController(IGameManager gameManager, ListingManager listingManager, IOptions<ServerConfig> serverConfig)
+    {
+        _gameManager = gameManager;
+        _listingManager = listingManager;
+        var config = serverConfig.Value;
+        _hostServer = HostServer.From(IPAddress.Parse(config.ResolvePublicIp()), config.PublicPort);
+    }
+
+    /// <summary>
+    /// Get a list of active games.
+    /// </summary>
+    /// <param name="mapId">Maps that are requested.</param>
+    /// <param name="lang">Preferred chat language.</param>
+    /// <param name="numImpostors">Amount of impostors. 0 is any.</param>
+    /// <returns>An array of game listings.</returns>
+    [HttpGet]
+    public IActionResult Index(int mapId, GameKeywords lang, int numImpostors)
+    {
+        var listings = _listingManager.FindListings(HttpContext, mapId, numImpostors, lang);
+        return Ok(listings.Select(GameListing.From));
+    }
+
+    /// <summary>
+    /// Get the address a certain game is hosted at.
+    /// </summary>
+    /// <param name="gameId">The id of the game that should be retrieved.</param>
+    /// <returns>The server this game is hosted on.</returns>
+    [HttpPost]
+    public IActionResult Post(int gameId)
+    {
+        var code = new GameCode(gameId);
+        var game = _gameManager.Find(code);
+
+        // If the game was not found, print an error message.
+        if (game == null)
+        {
+            return NotFound(new MatchmakerResponse(new MatchmakerError(DisconnectReason.GameMissing)));
+        }
+
+        return Ok(HostServer.From(game.PublicIp));
+    }
+
+    /// <summary>
+    /// Get the address to host a new game on.
+    /// </summary>
+    /// <returns>The address of this server.</returns>
+    [HttpPut]
+    public IActionResult Put()
+    {
+        return Ok(_hostServer);
+    }
+
+    private static uint ConvertAddressToNumber(IPAddress address)
+    {
+#pragma warning disable CS0618 // Among Us only supports IPv4
+        return (uint)address.Address;
```

Comment:
.Address [throws if provided an IPv6 address](https://learn.microsoft.com/en-us/dotnet/api/system.net.ipaddress.address?view=net-7.0#exceptions), maybe check if it is an IPv4 address using AddressFamily and if not, log something helpful instead of throwing an exception

Reply:
It would throw in https://github.com/Impostor/Impostor/blob/7b341bea147b763ef879e13e82c072dcf5ae6e25/src/Impostor.Api/Utils/IpUtils.cs#L32-L36 before it gets to this point